python/python/tests/test_multi_bucket.py

```diff
@@ -0,0 +1,313 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright The Lance Authors
+
+"""
+Tests for multi-bucket dataset functionality.
```

Comment:
Can we change the general naming to "multi-base"? Because bucket is a very S3 specific thing and we want to use Lance terminologies to be generic